pkg/model/engine.go

```diff
@@ -13,9 +13,7 @@ const (
 	EngineNoop
 	EngineDocker
 	EngineWasm
-	EngineLanguage   // wraps python_wasm
-	EnginePythonWasm // wraps docker
-	engineDone       // must be last
+	engineDone // must be last
```

Comment:
Safe? Yea, as long as we don't plan to cut a release with this change right now.
These (enums/constants) will all go away when https://github.com/bacalhau-project/bacalhau/issues/2542 lands, which I plan to follow this work with. I am landing this PR as a standalone change to prevent the following work from being hard to review as it would change a lot if everything was made in one change.

Also notice that this branch is based on https://github.com/bacalhau-project/bacalhau/pull/2550 (which I would also love your review on!), which is targeting `dev1`, a branch I made off `main` following the processes I proposed [here](https://www.notion.so/expanso/Plugable-Engine-Work-lane-bb73740fb34342f19d8c1e2e55ef749b?pvs=4#86ad78b263504f9084674673fc382ec3)

Reply:
The problem is existing compute nodes will continue to publish `EngineLanguage` in their `NodeInfo` message, which will cause `ParseEngine` to fail. This has happened in a previous release which was rolledback.

Reply:
But as long as we don't cut a release with this right now that isn't a problem, right? 

Things will break even more when we make the engines pluggable and replace these enums with strings. Would we prefer to land the removal of these executors with the pluggable engine change? It will make for a larger PR to review going that route, which is why I've made this a separate change.

Reply:
I suggest to gracefully handle unknown engines while processing NodeInfo messages instead of throwing errors and failing to register the compute node. This would be the case in the future when we switch to strings instead of enums for engine types.

So my recommendation for now is to change `ParseEngine` to return `unknown` instead of failing. I don't think you need to change `IsValidEngine`, but worth confirming